src/citrine/resources/file_link.py

```diff
@@ -547,10 +575,77 @@ def delete(self, file_link: FileLink):
             Resource referencing the external file.
 
         """
+        if self._is_external_url(file_link.url):
+            raise ValueError(f"Only local resources can be deleted; passed URL {file_link.url}")
+        if not self._validate_filelink_url(file_link.url):
+            raise ValueError(f"URL was malformed for local resources; passed URL {file_link.url}")
         split_url = file_link.url.split('/')
-        assert split_url[-2] == 'versions' and split_url[-4] == 'files', \
-            "File URL is expected to end with '/files/{{file_id}}/version/{{version id}}', " \
-            "but FileLink instead has url {}".format(file_link.url)
         file_id = split_url[-3]
         data = self.session.delete_resource(self._get_path(file_id))
         return Response(body=data)
+
+    def _resolve_file_link(self, identifier: Union[str, UUID]) -> FileLink:
+        """Generate the FileLink object referenced by the passed argument."""
+        if isinstance(identifier, FileLink):
```

Comment:
Isn't a FileLink outside the expected types of {str, UUID}? Is this just a "if they passed us a file link" exception case?

Reply:
No, I just failed to implement a consistent typing here because elsewhere I decided I wanted to avoid testing type in every method that leverages this one.

Reply:
Added `FileLink` to the argument set.